backends/dialogs/gtk/gtk-dialogs.cpp

```diff
@@ -35,25 +35,19 @@
 
 #include <gtk/gtk.h>
 
-Common::DialogManager::DialogResult GtkDialogManager::showFileBrowser(const char *title, Common::FSNode &choice, bool isDirBrowser) {
+Common::DialogManager::DialogResult GtkDialogManager::showFileBrowser(const Common::U32String &title, Common::FSNode &choice, bool isDirBrowser) {
 	if (!gtk_init_check(NULL, NULL))
 		return kDialogError;
 
 	DialogResult result = kDialogCancel;
 
-	// Get current encoding
-	Common::String guiEncoding = "ASCII";
-#ifdef USE_TRANSLATION
-	guiEncoding = TransMan.getCurrentCharset();
-#endif
-	Common::Encoding utf8("utf-8", guiEncoding);
+	Common::U32String choose = _("Choose");
+	Common::U32String cancel = _("Cancel");
 
 	// Convert labels to UTF-8
-	char *utf8Title = utf8.convert(title, strlen(title));
-	Common::String choose = _("Choose");
-	char *utf8Choose = utf8.convert(choose.c_str(), choose.size());
-	Common::String cancel = _("Cancel");
-	char* utf8Cancel = utf8.convert(cancel.c_str(), cancel.size());
+	char *utf8Title = (char *)title.encode().c_str();
+	char *utf8Choose = (char *)choose.encode().c_str();
+	char* utf8Cancel = (char *)cancel.encode().c_str();
```

Comment:
Even though it makes sense that a copy is generated by `encode()` and this shouldn't work, it still did. I manually browsed the memory and the titles were present. Still, to be on the safe side, I have changed it around.

Also, this helped me point out another obvious wrong usage, which was a memory corruption because I was freeing memory that wasn't mine. Have fixed them both in a new commit.